server/app/jobs/deployment_cleanup_job.rb

```diff
@@ -0,0 +1,32 @@
+require_relative '../services/logging'
+
+class DeploymentCleanupJob
+  include Celluloid
+  include CurrentLeader
+  include Logging
+
+  def initialize(perform = true)
+    async.perform if perform
+  end
+
+  def perform
+    info 'starting to cleanup old deployments'
+    loop do
+      sleep 5.minute.to_i
+      if leader?
+        destroy_old_deployments
+      end
+    end
+  end
+
+  def destroy_old_deployments
+    GridService.each do |s|
+      return if s.grid_service_deploys.finished.count <= 100
```

Comment:
The `return` is breaking the entire `each` loop. This should be `next`... or split into a separate method like `cleanup_old_deployments(s) if s.grid_service_deploys.finished.count <= 100`

Reply:
Fix + specs?